nixos/modules/services/networking/kresd.nix

```diff
@@ -129,12 +129,18 @@ in {
     systemd.services."kresd@".serviceConfig = {
       ExecStart = "${package}/bin/kresd --noninteractive "
         + "-c ${package}/lib/knot-resolver/distro-preconfig.lua -c ${configFile}";
+      # Ensure /run/knot-resolver exists
+      RuntimeDirectory = "knot-resolver";
+      RuntimeDirectoryMode = "0750";
+      # Ensure /var/lib/knot-resolver exists
+      StateDirectory = "knot-resolver";
+      StateDirectoryMode = "0750";
       # Ensure correct ownership in case UID or GID changes.
       CacheDirectory = "knot-resolver";
       CacheDirectoryMode = "0750";
```

Comment:
These permissions should all be 0770 (to be consistent with what's in the tmpfiles file)

Reply:
Done.